upgrade/helpers/rhevm.py

```diff
@@ -179,3 +188,99 @@ def wait_till_rhevm_instance_status(instance_name, status, timeout=5):
+                                                                new_template))
+                get_client.disconnect()
+        except Exception as ex:
+            logger.error('Failed to Create Template from VM:\n%s' % str(ex))
+            get_client.disconnect()
+    else:
+        get_client.disconnect()
+        logger.error('Low Storage cannot proceed or VM not found')
+
+
+# Fabric task
+def validate_and_create_product_templates():
+    """Task to do a sanity check on the satellite and capsule and then
+    create their templates after z-stream upgrade
+
+    Environment variables required to run upgrade on RHEVM Setup and will be
+    fetched from Jenkins:
+    ----------------------------------------------------------------------
+
+    RHEV_SAT_HOST
+        The rhevm satellite hostname to run upgrade on
+    RHEV_CAP_HOST
+        The rhevm capsule hostname to run upgrade on
+    RHEV_STORAGE
+        The storage domain on the rhevm used to create templates
+    RHEV_SAT_IMAGE
+        The satellite Image from which satellite instance will be created
+    RHEV_CAP_IMAGE
+        The capsule Image from which capsule instance will be created
+    RHEV_SAT_INSTANCE
+        The satellite instance name in rhevm of which template is to be
+        created, generally the upgraded box
+    RHEV_CAP_INSTANCE
+        The capsule instance name in rhevm of which template is to be
+        created, generally the upgraded box
+    """
+    # Get the instances name, specified in the jenkins job
+    sat_instance = os.environ.get('RHEV_SAT_INSTANCE')
+    cap_instance = os.environ.get('RHEV_CAP_INSTANCE')
+    cluster = 'Default'
+    storage = os.environ.get('RHEV_STORAGE')
+    if sat_instance and cap_instance:
+        sat_host = os.environ.get('RHEV_SAT_HOST')
+        new_sat_template = os.environ.get('RHEV_SAT_IMAGE') + "_new"
+        cap_host = os.environ.get('RHEV_CAP_HOST')
+        new_cap_template = os.environ.get('RHEV_CAP_IMAGE') + "_new"
+        if check_necessary_env_variables_for_upgrade('capsule'):
+            execute(check_ntpd, host=sat_host)
+            execute(katello_restart, host=sat_host)
+            execute(check_capsule, cap_host, host=sat_host)
+            execute(check_ntpd, host=cap_host)
+            execute(katello_restart, host=cap_host)
+            try:
+                thread.start_new_thread(create_rhevm_template,
+                                        (sat_instance,
+                                         cluster,
+                                         new_sat_template,
+                                         storage
+                                         ))
+                thread.start_new_thread(create_rhevm_template,
+                                        (cap_instance,
+                                         cluster,
+                                         new_cap_template,
+                                         storage))
+            except:
+                    raise
```

Comment:
I think you can just remove the `try` block and call `thread.start_new_thread` directly as the above `try` is useless, it is just re-raising. 

Remove it and then watch the errors happening, when the first error comes you will know which exception is more common and can decide what to do.